packages/composite-checkout/src/components/checkout-steps.js

```diff
@@ -217,7 +217,7 @@ export function CheckoutStepBody( {
 					title={ titleContent }
 					isActive={ isStepActive }
 					isComplete={ isStepComplete }
-					onEdit={ isStepComplete && goToThisStep ? goToThisStep : null }
+					onEdit={ isStepComplete && goToThisStep && ! isStepActive ? goToThisStep : null }
```

Comment:
I think this is where we'd disable editing when the form is in submitting state. If you think it makes sense we can include that change here.